### PR TITLE
fix(setup): refresh PATH from registry before tool checks

### LINKED ISSUE #129: chore(setup): post-install.ps1 の Test-CommandExists が Phase 2 直後のインストール結果を古い PATH で見落とす可能性
## Background

PR #127（CodeRabbit CLI installer、issue #126）のレビューで、
`Sync-CoderabbitCli` の `Get-Command git -ErrorAction SilentlyContinue`
チェックについて、次のシナリオが指摘された: クリーンマシンで
`boxstarter.ps1` を初回実行した場合、Phase 2 の WinGet
Configuration/import が `Git.Git` をインストールするが（
`configurations/packages.dsc.yaml` / `packages.import.json` の
full/min 両構成に含まれることを確認済み）、そのインストーラの
子プロセスは `boxstarter.ps1` 自身の親プロセスの `$env:Path` を
更新できない。Phase 5 の `Sync-CoderabbitCli` は同一プロセス内で
実行されるため、古い PATH のまま `Get-Command git` を評価してしまい、
Git が実際にインストール済みでも「見つからない」と誤判定して
CodeRabbit CLI のインストールをスキップする可能性がある。

同じ `Test-CommandExists` パターンは `libs/post-install.ps1` の
dotnet / vagrant / mkcert セクションでも使われており、これらも
Phase 2 の WinGet 構成に含まれる場合は同じタイミング問題を抱えうる。
`post-install.ps1` 自身のヘッダーコメントが
"idempotent and safe to re-run after a reboot" と明記しており、
Boxstarter の再実行（新しいプロセス、新しい PATH）でこの誤判定は
自己回復するため、PR #127 のスコープではこの指摘を
「有効だが本 PR の差分ではなく既存パターン」として見送った
（`Rejected` disposition、本 issue へのフォローアップを予告）。

Refs #126

## Proposed change

Phase 2 で新規インストールされたツールを Phase 5 が同一プロセス内で
確実に検出できるよう、ツール存在チェックの直前でマシンスコープ /
ユーザースコープの PATH をレジストリ（
`[System.Environment]::GetEnvironmentVariable('Path','Machine')` /
`'User'`）から再構築し、現在プロセスの `$env:Path` にマージする
共通ヘルパーを導入する。`libs/post-install.ps1` の
`Test-CommandExists` と `libs/coderabbit-cli-installer.ps1` の
`Sync-CoderabbitCli` の両方から呼べる形を想定。

## Acceptance criteria

- Phase 2 相当でインストールされたばかりのツールが、同一プロセスの
  Phase 5 相当のチェックで確実に検出されることを示すテスト
  （レジストリ読み取りをモック可能な形で）
- 既存の post-install ツールチェック（dotnet, vagrant, mkcert,
  CodeRabbit CLI 用 git）が新ヘルパーを使うよう統一されるか、
  統一しない場合はその判断根拠を issue コメントまたは PR 本文に
  記録する
- PSScriptAnalyzer が変更後のファイルに対して指摘を出さない
- `pre-push-validate` 相当のフルチェーンがクリーンな作業ツリーに
  対して exit 0 で完了する

## IDD ブートストラップ

marker prefix は `setup-windows`。実行フローは
[`docs/idd-workflow.md`](docs/idd-workflow.md) を参照。

---

_Autopilot suitability: 3 / 5 -- higher is more autopilot-suitable;
below the configured floor is human-oriented._

_Effort: M -- author-estimated size; a soft autopilot
selection tie-breaker only._

### PR DESCRIPTION
**Pull request by kurone-kito**
## Summary

- Add `libs/process-path-sync.ps1`, a shared `Sync-ProcessPath` helper that
  rebuilds the current process's `$env:Path` by merging the Machine- and
  User-scope PATH values read live from the registry (via a new,
  independently mockable `Get-RegistryPathValue` wrapper) on top of
  whatever is already in `$env:Path`, deduplicating case-/trailing-
  separator-insensitively.
- Call `Sync-ProcessPath` as the first statement inside
  `libs/post-install.ps1`'s `Test-CommandExists` and
  `libs/coderabbit-cli-installer.ps1`'s `Sync-CoderabbitCli`, so a tool
  installed earlier in the same PowerShell process (e.g. by WinGet
  Configuration in Phase 2) is reliably detected by a `Get-Command`-based
  check later in the same process (e.g. Phase 5's dotnet/vagrant/mkcert/
  git checks), instead of only after a reboot.
- Add `tests/powershell/process-path-sync.Tests.ps1` and extend
  `tests/powershell/coderabbit-cli-installer.Tests.ps1`'s `Sync-CoderabbitCli`
  block with a `Sync-ProcessPath` mock/assertion.

Closes #129

## Background

A child installer launched during Phase 2 (WinGet Configuration) only
ever persists PATH changes to the Machine/User registry — it cannot
reach back into `boxstarter.ps1`'s already-running process to update
that process's own `$env:Path`, and only a brand-new process picks up a
persisted PATH change. Phase 5 (`libs/post-install.ps1`'s
`Test-CommandExists` calls for dotnet/vagrant/mkcert, and
`libs/coderabbit-cli-installer.ps1`'s own `git` check inside
`Sync-CoderabbitCli`) runs in that same process, so a tool installed in
Phase 2 could be misreported as "not found" and its dependent step
skipped, until the next reboot/process restart. This was raised during
PR #127's review (issue #126) and deliberately deferred to this
follow-up issue.

## Judgment call: unification scope (acceptance criterion #2)

**Unified**: `Sync-ProcessPath` is called from both `Test-CommandExists`
(covering all three of its call sites — dotnet, vagrant, mkcert — without
touching each site individually) and `Sync-CoderabbitCli` (its `git`
check), because Phase 2's WinGet Configuration can install any of
dotnet/git/vagrant/mkcert equally — there's no technical reason to
special-case the `git` check.

**Not merged**: `Test-CommandExists` itself is not folded into
`Sync-CoderabbitCli`'s `git` check to create one single shared
existence-check function. `coderabbit-cli-installer.ps1`'s existing
docstring explains it intentionally keeps its `git` check independent of
`post-install.ps1`'s `Test-CommandExists`, so `Sync-CoderabbitCli` stays
testable via `tests/powershell/coderabbit-cli-installer.Tests.ps1` without
dot-sourcing `post-install.ps1`. Only the PATH-refresh mechanism is
unified; each file keeps its own existence-check implementation.

**Excluded**: `libs/unity-editor-installer.ps1`'s `Test-UnityCliAvailable`
and `libs/unity-cli-installer.ps1`'s own `Get-Command unity` checks. The
Unity CLI is never installed via Phase 2 WinGet Configuration (only
`Unity.UnityHub` — the GUI hub, not the `unity` CLI binary — is a
declared winget package); the CLI itself is installed via its own
direct-download `install.ps1`, and `libs/unity-cli-installer.ps1` already
ships its own dedicated `Add-UnityCliToProcessPath` process-PATH refresh
around that install call. So Unity doesn't have the stale-registry-vs-
process-PATH gap this issue addresses. `libs/strategy.ps1`'s
`Test-ConfigurationStrategy` also checks `Get-Command winget`, but that
check runs as Phase 2's own first gate — `winget` must already resolve
before Phase 2 (the phase this issue is about) can run at all, so it
isn't a "just installed earlier in this same process" case either.

## Notes on the new test

`tests/powershell/process-path-sync.Tests.ps1`'s end-to-end test creates
a real (temporary) executable to model a tool Phase 2 "just installed,"
then asserts via `$env:Path`'s string content — not a live `Get-Command`
round-trip — that the tool's directory is absent before `Sync-ProcessPath`
and present after. This mirrors the existing
`Add-CoderabbitCliToProcessPath`/`Add-UnityCliToProcessPath` Pester tests
in this repo, which likewise only assert on `$env:Path` content: on
Windows (this function's only real target), `$env:Path` and the OS-level
`PATH` variable `Get-Command` actually resolves against are the same
case-insensitive variable, but on Linux/macOS they are distinct,
case-sensitive variables — so a real `Get-Command` round-trip through
`$env:Path` would be meaningless on the non-Windows runner this repo's
Pester CI job (`ubuntu-latest`) actually executes on. A best-effort real
`Get-Command` assertion is still included, gated on `$IsWindows` (skipped
everywhere else, including this repo's CI).

## Test plan

- [x] `npx -y markdownlint-cli2 "**/*.md"` — clean
- [x] `npx -y cspell lint "**" --no-progress` — clean
- [x] `pwsh -c "Invoke-ScriptAnalyzer -Path . -Recurse -Settings ./PSScriptAnalyzerSettings.psd1 -EnableExit"` — 0 findings
- [x] `pwsh -c "Invoke-Pester -Path ./tests/powershell -CI"` — 143 passed, 0 failed
- [ ] CI (this PR's own checks)

## Summary by CodeRabbit

* **Bug Fixes**
  * Newly installed command-line tools are now detected immediately without restarting the current shell or process.
  * Installer and post-install checks refresh the active PATH before verifying required commands.
  * PATH synchronization preserves existing entries, removes duplicates, and consistently combines system and user settings.

* **Tests**
  * Added coverage for PATH merging, command detection, duplicate handling, ordering, environment restoration, and edge cases.

### CODE REVIEW

**Comment by kurone-kito** [comment]
advisory-wait-recovery: claude-code-TKOAM-PCWN-0060-44a061c7 9bfab5baf070cca8c0490bbb1edbda3a15ab7dfa 2026-09-01T06:25:56Z

**Comment by kurone-kito** [comment]
_claude-code-TKOAM-PCWN-0060-44a061c7: review triage snapshot — IDD automation marker. Do not edit._

File: tests/powershell/coderabbit-cli-installer.Tests.ps1 [hunk 1]
```diff
@@ -156,6 +156,30 @@ Describe 'Invoke-CoderabbitCliInstaller' {
...
+  It 'refreshes the process PATH even when git turns out not to be found' {
+    Mock Get-Command -ParameterFilter { $Name -eq 'git' } { $null }
+    Mock Invoke-CoderabbitCliInstaller { 0 }
+
+    Sync-CoderabbitCli -WarningAction SilentlyContinue
+
+    Should -Invoke Sync-ProcessPath -Times 1
   }
 
   It 'skips installation and does not touch PATH when git is not found' {
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply tests/powershell/coderabbit-cli-installer.Tests.ps1]: **Accepted** — the pre-existing test title claimed Sync-CoderabbitCli does not touch PATH when git is missing, but it now calls Sync-ProcessPath unconditionally as its first statement (issue #129), so the title was stale. Renamed to 'skips installation and does not add the CodeRabbit CLI install dir to PATH when git is not found', which is what the test body actually asserts (commit 2854444).

**Comment by kurone-kito** [comment]
advisory-wait-recovery: claude-code-TKOAM-PCWN-0060-44a061c7 28544443eca03ec8b60b60886e6bf55781b070eb 2026-09-01T06:37:15Z

**Comment by kurone-kito** [comment]
advisory-wait-recovery: claude-code-TKOAM-PCWN-0060-44a061c7 81a9f3125f1ed1ba3becfb09104e32ee0ec11462 2026-09-01T06:45:02Z

**Comment by kurone-kito** [comment]
_claude-code-TKOAM-PCWN-0060-44a061c7: review triage snapshot — IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
**Rejected** — coderabbitai[bot] did not review HEAD 81a9f3125f1ed1ba3becfb09104e32ee0ec11462 (review limit reached / rate limited); this is not a completed review (source: #issuecomment-5489799866)

**Comment by kurone-kito** [comment]
_claude-code-TKOAM-PCWN-0060-44a061c7: review triage snapshot — IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
**F4 Cleanup Evidence**

| Field                            | Value                                                  |
| --------------------------------- | ------------------------------------------------------ |
| Status                            | clean                                                   |
| Applied                           | 0                                                        |
| Failed                            | 0                                                        |
| Skipped                           | 11                                                       |
| Permission-blocked                | 0                                                        |
| Retry attempts (bound-exhausted)  | 1 (false)                                                |
| Notes                             | All 5 candidates from this run's dry-run were already minimized by the time `--apply` ran (a repository post-merge workflow, `github-actions[bot]`, converged them first -- not configured as a trusted marker actor in `.github/idd/config.json`, so this comment records this session's own trusted evidence per the duplicate-success-record rule). 2 review parents remain skipped (no associated review threads). |
